### PR TITLE
Merged ADMS changes for Nov. SDK release

### PR DESCRIPTION
**Pull request by gadongji**
## Description

 This is a merged PR for three SDK changes from ADMS side.
 Related features and corresponding swagger changes, SDK changes and internal items are:

AzureMySql connector:
[Swagger PR] https://github.com/Azure/azure-rest-api-specs/pull/1911
[Internal item] [97894](https://msdata.visualstudio.com/Azure%20Data%20Factory/_workitems/edit/97894)
[Previous SDK PR] https://github.com/Azure/azure-sdk-for-net/pull/3813

Dynamics, Salesforce and JSONFormat:
[Swagger PR] https://github.com/Azure/azure-rest-api-specs/pull/1850
                      https://github.com/Azure/azure-rest-api-specs/pull/1903
[Internal item] [97878](https://msdata.visualstudio.com/Azure%20Data%20Factory/_workitems/edit/97878)
[Previous SDK PR] https://github.com/Azure/azure-sdk-for-net/pull/3811

Support providing Salesforce passwords and security tokens as SecureStrings or as secrets in Azure Key Vault:
[Swagger PR] https://github.com/Azure/azure-rest-api-specs/pull/1917
[Internal item] [97519](https://msdata.visualstudio.com/Azure%20Data%20Factory/_workitems/edit/97519)
[Previous SDK PR] https://github.com/Azure/azure-sdk-for-net/pull/3816

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by bgold09 — APPROVED** [review]

File: src/SDKs/DataFactory/Management.DataFactory/changelog.md [hunk 1]
```diff
@@ -4,6 +4,14 @@
 
 ### Feature Additions
   * Cancel pipeline run api.
+  * Add AzureMySql linked service.
```
Comment (markcowl) [inline src/SDKs/DataFactory/Management.DataFactory/changelog.md]: You will need to update the version for these changes in your csproj file (0.3.0-preview or 0.2.1-preview).

File: src/SDKs/DataFactory/DataFactory.Tests/JsonSamples/DatasetJsonSamples.cs [hunk 2]
```diff
@@ -402,5 +402,71 @@ public class DatasetJsonSamples : JsonSampleCollection<DatasetJsonSamples>
     }
 }
 ";
+        [JsonSample]
```
Comment (markcowl) [inline src/SDKs/DataFactory/DataFactory.Tests/JsonSamples/DatasetJsonSamples.cs]: Thanks for the samples, but I don't see any actual test updates

**Review by markcowl — CHANGES_REQUESTED** [review]
You will need to update the version, also, where are the covering tests for the API changes?

File: src/SDKs/DataFactory/DataFactory.Tests/JsonSamples/DatasetJsonSamples.cs (hunk 2 above)
Comment (bgold09) [reply src/SDKs/DataFactory/DataFactory.Tests/JsonSamples/DatasetJsonSamples.cs]: these run automatically as part of `LinkedService_SerializationTest`, `Dataset_SerializationTest` and `Activity_SerializationTest`

Comment (bgold09) [reply src/SDKs/DataFactory/DataFactory.Tests/JsonSamples/DatasetJsonSamples.cs]: see the `ClassData` attribute on each

File: src/SDKs/DataFactory/Management.DataFactory/changelog.md (hunk 1 above)
Comment (bgold09) [reply src/SDKs/DataFactory/Management.DataFactory/changelog.md]: We'll be bumping the version in a separate change soon (to 0.2.1-preview), thanks for the reminder

**Review by markcowl — APPROVED** [review]

**Pull request merged**